src/cbc_sdk/workload/compliance_assessment.py

```diff
@@ -0,0 +1,465 @@
+            scan_schedule (str): The scan schedule, specified in RFC 5545 format. Example: "RRULE:FREQ=DAILY;BYHOUR=17;BYMINUTE=30;BYSECOND=0".
+            scan_timezone (str): The timezone in which the scan will run, specified as a timezone string. Example: "UTC".
+
+        Raises:
+            ApiError: If cb is not an instance of CBCloudAPI, or if scan_schedule or scan_timezone are not provided.
+
+        Returns:
+            The JSON response from the Carbon Black Cloud API as a Python dictionary.
+
+        Example:
+            >>> cb = CBCloudAPI(profile="example_profile")
+            >>> schedule = ComplianceBenchmark.set_compliance_schedule(cb, scan_schedule="RRULE:FREQ=DAILY;BYHOUR=17;BYMINUTE=30;BYSECOND=0", scan_timezone="UTC")
+            >>> print(schedule)
+        """
+        if cb.__class__.__name__ != "CBCloudAPI":
+            message = "cb argument should be instance of CBCloudAPI."
+            message += "\nExample:\ncb = CBCloudAPI(profile='example_profile')"
+            message += "\nComplianceBenchmark.set_compliance_schedule(cb, scan_schedule='RRULE:FREQ=DAILY;BYHOUR=17;BYMINUTE=30;BYSECOND=0', scan_timezone='UTC')"
+            raise ApiError(message)
+        if not scan_schedule or not scan_timezone or scan_schedule == "" or scan_timezone == "":
+            raise ApiError(
+                "scan_schedule and scan_timezone are required. http://developer.carbonblack.com/reference/carbon-black-cloud/cb-liveops/latest/livequery-api/#recurrence-rules")
+
+        args = {"scan_schedule": scan_schedule, "scan_timezone": scan_timezone}
+        url = f"/compliance/assessment/api/v1/orgs/{cb.credentials.org_key}/settings"
+
+        return cb.put_object(url, body=args).json()
+
+    def search_rules(self, rule_id=None):
+        """
+        Fetches compliance rules associated with the benchmark set.
+
+        Args:
+            rule_id (str, optional): The rule ID to fetch a specific rule. Defaults to None.
+
+        Yields:
+            ComplianceBenchmark: A Compliance object representing a compliance rule.
+
+        Example:
+            >>> cb = CBCloudAPI(profile="example_profile")
+            >>> benchmark_sets = cb.select(ComplianceBenchmark)
+            >>> # To return all rules within a benchmark set, leave search_rules empty.
+            >>> rules = list(benchmark_sets[0].search_rules())
+            >>> print(*rules)
+            >>> # To return a single rule document, add the rule ID.
+            >>> rule = list(benchmark_sets.search_rules('00869D86-6E61-4D7D-A0A3-6F5CDE2E5753'))
+            >>> print(rule)
+        """
+        if rule_id is not None:
+            self._rule_id = rule_id
+            url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/rules/{rule_id}"
+            result = self._cb.get_object(url)
+            yield ComplianceBenchmark(self._cb, initial_data=result)
```

Comment:
It seems odd to return ComplianceBenchmark as that is a different data model. We should create a sub class for ComplainceBenchmarkRule instead ComplianceBenchmark as its possible to update the rule to be Enabled/Disabled.